client/api/filesystem/v1alpha1/api.proto

```diff
@@ -4,25 +4,156 @@ package v1alpha1;
+    // and target path on the host filesystem.
+    rpc LinkPath(LinkPathRequest) returns (LinkPathResponse) {}
+}
+
+// Context of the paths used for path prefix validation
+enum PathContext {
+    // Indicates the kubelet-csi-plugins-path parameter of csi-proxy be used as the path context
+    PLUGIN = 0;
+    // Indicates the kubelet-pod-path parameter of csi-proxy be used as the path context
+    CONTAINER = 1;
 }
 
 message PathExistsRequest {
-    // The path to check in the host filesystem.
+    // The path whose existence we want to check in the host filesystem.
     string path = 1;
+
+    // Context of the path parameter.
+    // This is used to determine the root for relative path parameters
+    PathContext context = 2;
 }
 
 message PathExistsResponse {
-    bool success = 1;
+    // The Windows API error code observed (if any)
+    api.WindowsApiError api_error = 2;
+
+    // Indicates whether the path in PathExistsRequest exists in the host
+    bool exists = 3;
+}
 
-    // present iff success is false
-    api.CmdletError cmdlet_error = 2;
+message MkdirRequest {
+    // The path to create in the host filesystem.
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    // Non-existent parent directories in the path will be automatically created.
+    // Directories will be created with Read and Write privileges of the Windows
+    // User account under which csi-proxy is started (typically LocalSystem).
+    //
+    // Restrictions:
+    // If an absolute path (indicated by a drive letter prefix: e.g. "C:\") is passed,
+    // depending on the context parameter of this function, the path prefix needs 
+    // to match the paths specified either as kubelet-csi-plugins-path
+    // or as kubelet-pod-path parameters of csi-proxy.
+    // If a relative path is passed, depending on the context parameter of this 
+    // function, the path will be considered relative to the path specified either as 
+    // kubelet-csi-plugins-path or as kubelet-pod-path parameters of csi-proxy.
+    // The path parameter cannot already exist on host filesystem.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
+    // All directory separators need to be backslash character: "\".
+    // Characters: .. / : | ? * in the path are not allowed.
+    // Maximum path length will be capped to 260 characters.
+    string path = 1;
+
+    // Context of the path parameter.
+    // This is used to [1] determine the root for relative path parameters
+    // or [2] validate prefix for absolute paths (indicated by a drive letter 
+    // prefix: e.g. "C:\")
+    PathContext context = 2;
+}
+
+message MkdirResponse {
+    // The Windows API error code observed (if any)
+    api.WindowsApiError api_error = 1;
```

Comment:
Same remark as above, not sure this should be a `WindowsApiError`; maybe a `CmdletError` if the implementation shells out to `New-Item`, or just a string error message if using `os.MkdirAll`?